packages/ra-core/src/controller/list/useListController.spec.tsx

```diff
@@ -585,4 +586,81 @@ describe('useListController', () => {
             expect(authProvider.checkAuth).not.toHaveBeenCalled();
         });
     });
+
+    describe('onSelectAll', () => {
+        const dataProvider = testDataProvider({
+            getList: jest.fn().mockImplementation((_resource, params) =>
+                Promise.resolve({
+                    data: [{ id: 0 }, { id: 1 }].slice(
+                        0,
+                        params.pagination.perPage
+                    ),
+                    total: 2,
+                })
+            ),
+        });
+        const Children = params => (
+            <div>
+                <button onClick={() => params.onSelectAll()}>Select All</button>
+                <button onClick={() => params.onSelectAll({ limit: 1 })}>
+                    Limited Select All
+                </button>
+                <button onClick={() => params.onSelect([0])}>Select 0</button>
+                <span>children</span>
+            </div>
+        );
```

Comment:
Same as `useInfiniteListController`, you should update the `Basic` story without a `children` prop

Reply:
I need to pass a direct children of the `useListController`  to be able to test if the list has been called with an object X or Y.

If you prefer I can add the `Select item 0` button to the story and change this `Children` to:
```tsx
const Children = params => <List {...params} />;
```